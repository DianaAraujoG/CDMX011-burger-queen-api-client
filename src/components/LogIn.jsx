import React, { Fragment, useState} from 'react';
import logo from '../assets/Burger-Queen-logo.png';
import { signInWithEmailAndPassword } from 'firebase/auth';
import { auth } from '../FirebaseConfig';
import { Link , useNavigate } from 'react-router-dom';
import './style/Style.css';
// import Swal from 'sweetalert2';
// import Errors from './errors';

function LogIn (){
    const [loginMail, setLoginMail] = useState('');
    const [loginPassword, setLoginPassword] = useState('');
    const navigate = useNavigate();

    const logInPromise = async (event) =>{
        event.preventDefault();
        try{
            const user = await signInWithEmailAndPassword(
                auth,
                loginMail,
                loginPassword
            );
            console.log(user);
            if(user){
               navigate("/menu");
            }
              
        }catch(error){
            // Swal.fire
            console.log(error.code);
        }
    }

    return(
        <Fragment>
          <main>
          <div className= 'container'><img src={logo} className="App-logo" alt="logo" /></div>
            <form>
                <h1 className ='h1Form'>Iniciar sesi칩n</h1>
                <label htmlFor= 'username-mail'> Correo Electr칩nico: </label>
                <input
                 id ='username-mail'
                 type='email'
                 placeholder='ejemplo@ejemplo.com'
                 aria-label="mail"
                 onChange={(event) => {
                    setLoginMail(event.target.value);
                  }}></input>
                 <label htmlFor= 'username-password'> Contrase침a: </label>
                 <input
                 id= 'username-password'
                 type='password'
                 placeholder='Ingrese su contrase침a'
                 onChange={(event) => {
                    setLoginPassword(event.target.value);
                  }}></input>
                 <button 
                 className= 'buttonForm'
                 type='submit'
                 onClick={logInPromise}>Enviar</button>
            </form>
            
          </main>
          <div className='link'><Link to = '/register' className='changeRoute'>Registrate</Link></div>
        </Fragment>    
    )
}

export default LogIn