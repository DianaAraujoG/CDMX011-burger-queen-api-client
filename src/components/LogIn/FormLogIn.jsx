import React, { useState } from 'react';


const FormLogIn = ({saveData}) =>{

const [loginMail, setLoginMail] = useState('');
const [loginPassword, setLoginPassword] = useState('');

    return(
        <form>
            <h1 className ='h1Form'>Iniciar sesi칩n</h1>
            <label htmlFor= 'username-mail'> Correo Electr칩nico: </label>
            <input
             id ='username-mail'
             type='email'
             placeholder='ejemplo@ejemplo.com'
             aria-label="mail"
             onChange={(event) => {
                setLoginMail(event.target.value);
              }}></input>
             <label htmlFor= 'username-password'> Contrase침a: </label>
             <input
             id= 'username-password'
             type='password'
             placeholder='Ingrese su contrase침a'
             onChange={(event) => {
                setLoginPassword(event.target.value);
              }}></input>
             <button 
             className= 'buttonForm'
             type='submit'
             onClick={(event) => {event.preventDefault();
             saveData(loginMail,loginPassword)}}>Enviar</button>
        </form>
    )
};

export default FormLogIn