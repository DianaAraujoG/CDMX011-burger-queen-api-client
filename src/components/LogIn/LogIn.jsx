import React, { Fragment } from 'react';
import logo from '../../assets/Burger-Queen-logo.png';
import { signInWithEmailAndPassword } from 'firebase/auth';
import { auth } from '../../FirebaseConfig';
import { Link , useNavigate } from 'react-router-dom';
import FormLogIn  from './FormLogIn'
import '../style/Style.css';

// import Errors from './errors';

function LogIn (){
    
    const navigate = useNavigate();

    const logInPromise = async (loginMail, loginPassword) =>{
        try{
            const user = await signInWithEmailAndPassword(
                auth,
                loginMail,
                loginPassword
            );
            console.log(user);
            if(user){
               navigate("/menu");
            }
              
        }catch(error){
            console.log(error.code);
        }
    }

    return(
        <Fragment>
          <main>
          <div className= 'container'><img src={logo} className="App-logo" alt="logo" /></div>
            <FormLogIn saveData={logInPromise}></FormLogIn>
          </main>
          <div className='link'><Link to = '/register' className='changeRoute'>Registrate</Link></div>
        </Fragment>    
    )
}

export default LogIn