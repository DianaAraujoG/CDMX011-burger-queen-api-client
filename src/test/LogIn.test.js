// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import LogIn from '../components/LogIn';
import { MemoryRouter } from 'react-router-dom';

describe('Test LogIn', () => {
    test('should render', () => {
      expect(render(<MemoryRouter><LogIn ></LogIn></MemoryRouter>)).toMatchSnapshot();
    }); 
    test("should render the basic fields", () => {
      render(<MemoryRouter><LogIn /></MemoryRouter>);
      expect(screen.getByLabelText('Correo Electrónico:')
      ).toBeInTheDocument();
      expect(screen.getByLabelText('Contraseña:')
      ).toBeInTheDocument();
    })
})