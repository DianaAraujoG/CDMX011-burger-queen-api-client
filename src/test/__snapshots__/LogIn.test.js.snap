// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test LogIn should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main>
        <div
          class="container"
        >
          <img
            alt="logo"
            class="App-logo"
            src="Burger-Queen-logo.png"
          />
        </div>
        <form>
          <h1
            class="h1Form"
          >
            Iniciar sesión
          </h1>
          <label
            for="username-mail"
          >
             Correo Electrónico: 
          </label>
          <input
            aria-label="mail"
            id="username-mail"
            placeholder="ejemplo@ejemplo.com"
            type="email"
          />
          <label
            for="username-password"
          >
             Contraseña: 
          </label>
          <input
            id="username-password"
            placeholder="Ingrese su contraseña"
            type="password"
          />
          <button
            class="buttonForm"
            type="submit"
          >
            Enviar
          </button>
        </form>
      </main>
      <div
        class="link"
      >
        <a
          class="changeRoute"
          href="/register"
        >
          Registrate
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <main>
      <div
        class="container"
      >
        <img
          alt="logo"
          class="App-logo"
          src="Burger-Queen-logo.png"
        />
      </div>
      <form>
        <h1
          class="h1Form"
        >
          Iniciar sesión
        </h1>
        <label
          for="username-mail"
        >
           Correo Electrónico: 
        </label>
        <input
          aria-label="mail"
          id="username-mail"
          placeholder="ejemplo@ejemplo.com"
          type="email"
        />
        <label
          for="username-password"
        >
           Contraseña: 
        </label>
        <input
          id="username-password"
          placeholder="Ingrese su contraseña"
          type="password"
        />
        <button
          class="buttonForm"
          type="submit"
        >
          Enviar
        </button>
      </form>
    </main>
    <div
      class="link"
    >
      <a
        class="changeRoute"
        href="/register"
      >
        Registrate
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
